api/nakadi-event-bus-api.yaml

```diff
@@ -904,6 +1187,23 @@ definitions:
         description: |
           Offset of the event being pointed to.
 
+  CursorCommitResult:
+    description: |
+      The result of single cursor commit. Holds a cursor itself and a result value.
+    required:
+      - cursor
+      - result
+    properties:
+      cursor:
+        $ref: '#/definitions/Cursor'
+      result:
+        type: string
+        description: |
+          The result of cursor commit.
+          - `committed`: cursor was successfully committed
+          - `outdated`: there already was more recent (or the same) cursor committed, so the current one was not
+            committed as it is outdated
+
   EventStreamBatch:
```

Comment:
I think the batch structure can be extended to publish status with a `stream_status` object or some such. For example, this API causes empty batches to occur because `window_size` was hit, but that can also occur because there's no data to consume and the client has no way of knowing what's going on.